src/cursor/cur_backup.c

```diff
@@ -18,6 +18,84 @@ static int __backup_stop(WT_SESSION_IMPL *, WT_CURSOR_BACKUP *);
 #define WT_CURSOR_BACKUP_CHECK_STOP(cursor) \
     WT_ERR(F_ISSET(((WT_CURSOR_BACKUP *)(cursor)), WT_CURBACKUP_FORCE_STOP) ? EINVAL : 0);
 
+/*
+ * __wt_backup_open --
+ *     Restore any incremental backup information. We use the metadata's block information as the
+ *     authority on whether incremental backup was in use on last shutdown.
+ */
+int
+__wt_backup_open(WT_SESSION_IMPL *session)
+{
+    WT_BLKINCR *blkincr;
+    WT_CONFIG blkconf;
+    WT_CONFIG_ITEM b, k, v;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    uint64_t i;
+    char *config;
+
+    config = NULL;
+    i = 0;
+    conn = S2C(session);
+    WT_RET(__wt_metadata_search(session, WT_METAFILE_URI, &config));
+    WT_ERR(__wt_config_getones(session, config, "checkpoint_backup_info", &v));
+    __wt_config_subinit(session, &blkconf, &v);
+    /*
+     * Walk each item in the metadata and set up our last known global incremental information.
+     */
+    F_CLR(conn, WT_CONN_INCR_BACKUP);
+    while (__wt_config_next(&blkconf, &k, &v) == 0) {
+        if (i >= WT_BLKINCR_MAX)
+            break;
```

Comment:
Since this is durably stored, I was thinking that if we ever increase to store more, and then later store fewer the metadata would have more information than we can handle right now. Or we might someday want to configure how many. For now I will make it an assert.